lib/serviceClient.ts

```diff
@@ -358,15 +358,20 @@ export function serializeRequestBody(serviceClient: ServiceClient, httpRequest:
   }
 }
 
-function createDefaultRequestPolicyFactories(credentials: ServiceClientCredentials | undefined, options: ServiceClientOptions, userAgentInfo: string[]): RequestPolicyFactory[] {
+function createDefaultRequestPolicyFactories(credentials: ServiceClientCredentials | RequestPolicyFactory | undefined, options: ServiceClientOptions, userAgentInfo: string[]): RequestPolicyFactory[] {
   const factories: RequestPolicyFactory[] = [];
 
   if (options.generateClientRequestIdHeader) {
     factories.push(generateClientRequestIdPolicy(options.clientRequestIdHeaderName));
   }
 
   if (credentials) {
-    factories.push(signingPolicy(credentials));
+    const crendentialsPolicyFactory = credentials as RequestPolicyFactory;
+    if (!crendentialsPolicyFactory || !crendentialsPolicyFactory.create) {
```

Comment:
`!credentialsPolicyFactory` will always be false, because you just checked `credentials` a couple of lines above. In C# the `as` keyword will return null if the value is not a `RequestPolicyFactory`, but in TypeScript it can be considered a plain-old C-style cast.

Reply:
Please remove `!credentialsPolicyFactory`.

Reply:
The nicest thing might be to replace with a **user defined type guard**
```ts
function isRequestPolicyFactory(obj: any): obj is RequestPolicyFactory {
  return typeof obj.create === "function";
}
```